File: private key encryptions/java/DESEncryption.java
// data encryption standard
// older and less secure
// block cipher with a 56-bit key


import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

public class DESEncryption
  {
    public static byte[] encrypt(byte[] plaintext, byte[] key) throws Exception
    {
      Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
      SecretKeySpec keySpec = new SecretKeySpec(key, "DES");
      cipher.init(Cipher.ENCRYPT_MODE, keySpec);
      byte[] ciphertext = cipher.doFinal(plaintext);
      return ciphertext;
    }
    public static byte[] decrypt(byte[] ciphertext, byte[] key) throws Exception
    {
      Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
      SecretKeySpec keySpec = new SecretKeySpec(key, "DES");
      cipher.init(Cipher.DECRYPT_MODE, keySpec);
      byte[] plaintext = cipher.doFinal(ciphertext);
      return plaintext;
    }
  }

File: private key encryptions/python/DESEncryption.py
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes 

def encrypt_des(plaintext, key):
  ciper = Cipher.new(algorithms.DES(key), modes.ECB())
  ciphertext = cipher.encrypt(plaintext)
  return ciphertext 

def decrypt_des(ciphertext, key):
  cipher = Cipher.new(algorithms.DES(key), modes.ECB())
  plaintext = cipher.decrypt(ciphertext)
  return plaintext

# example usage
plaintext = b"Hellow, world!"
key = b'secretkey' # 8 byte key

ciphertext = encrypt_des(plaintext, key)
decrypted_plaintext = decrypt_des(ciphertext, key)

print(decrypted_plaintext)
